### ISSUE TITLE
Problem verifying bridge methods

### ISSUE THREAD

**Issue by fbaro** [mockito/mockito#304]
Hello, I think there might be a bug in mockito when verifying bridge methods.
The bug is present both in version 2.0.31-beta, and in version 1.10.19.

```java
import java.io.Serializable;
import org.junit.Test;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

public class MockitoBridgeTest {

    @Test
    public void test() {
        Extended ext = mock(Extended.class);
        ext.test("123");
        verify(ext).test("123"); // Passes
        ((Base<String>)ext).test("456");
        verify(ext).test("456"); // Fails
    }

    public interface Base<T extends Serializable> {
        int test(T value);
    }

    public interface Extended extends Base<String> {
        @Override
        int test(String value);
    }

}

**Comment by raphw** [comment]
The problem here is the following: Byte Buddy does not intercept bridge methods but assumes that these methods should be dispatched *as they are*. Therefore, the `Method` that relates to this interception is the actual method, i.e. `test(String)` and not `test(Serializable)` which is only added as a bridge.

Mockito on the other hand picks up the invoked method from the call site which is `test(Serializable)`. Instead, Mockito would need to resolve the method to its actual target in order to validate that this method is a bridge for `test(String)`.

I am going to look into how this can be added to Mockito.

**Comment by fbaro** [comment]
Since somebody else has this issue, I found a workaround to make the test pass:

```java

        ((Base<String>)ext).test("456");
        verify(ext).test("456"); // Fails
        ((Base<String>)verify(ext)).test("456"); // Passes (workaround)

**Comment by raphw** [comment]
@bric3 Would it be a feasible solution if I relied on Byte Buddy APIs to resolve this? I implemented a rather elaborate algorithm that I have tried to port to Mockito. However, its not as easy as I thought since the algorithm and related data structure are quite bound into the Byte Buddy APIs. I would of course add an interface that does not rely on Byte Buddy for querying this information but asking the library to do it would be the most straight-forward solution.
